### PR TITLE
help migrate more patches

### PR DESCRIPTION
**Pull request by Ushie**

### CODE REVIEW

File: src/main/kotlin/app/revanced/patches/youtube/misc/playertype/PlayerTypeHookPatch.kt [hunk 1]
```diff
@@ -1,25 +1,28 @@
...
 import app.revanced.patches.youtube.misc.playertype.fingerprint.VideoStateFingerprint
 import com.android.tools.smali.dexlib2.iface.instruction.ReferenceInstruction
 
-@Name("Player type hook")
-@Description("Hook to get the current player type and video playback state.")
-@DependsOn([IntegrationsPatch::class])
-class PlayerTypeHookPatch : BytecodePatch(
-    listOf(PlayerTypeFingerprint, VideoStateFingerprint)
+@Patch(
+    name = "Player type hook",
```
Comment (oSumAtrIX) [inline src/main/kotlin/app/revanced/patches/youtube/misc/playertype/PlayerTypeHookPatch.kt:16]:
```suggestion
```

File: src/main/kotlin/app/revanced/patches/youtube/misc/settings/SettingsPatch.kt [hunk 2]
```diff
@@ -1,39 +1,64 @@
...
 import com.android.tools.smali.dexlib2.util.MethodUtil
 import java.io.Closeable
 
-@DependsOn([IntegrationsPatch::class, SettingsResourcePatch::class])
-@Name("Settings")
-@Description("Adds settings for ReVanced to YouTube.")
-class SettingsPatch : BytecodePatch(
-    listOf(LicenseActivityFingerprint, SetThemeFingerprint)
+@Patch(
+    name = "Settings",
```
Comment (oSumAtrIX) [inline src/main/kotlin/app/revanced/patches/youtube/misc/settings/SettingsPatch.kt:22]:
```suggestion
```

File: src/main/kotlin/app/revanced/patches/youtube/misc/settings/SettingsResourcePatch.kt [hunk 3]
```diff
@@ -1,24 +1,73 @@
...
+)
+object SettingsResourcePatch : AbstractSettingsResourcePatch(
     "revanced_prefs",
     "settings"
 ) {
+    // Used for a fingerprint from SettingsPatch.
+    internal var appearanceStringId = -1L
+
+    // if this is not null, all intents will be renamed to this
+    var overrideIntentsTargetPackage: String? = null
```
Comment (oSumAtrIX) [inline src/main/kotlin/app/revanced/patches/youtube/misc/settings/SettingsResourcePatch.kt:27]: Should be internal.

File: src/main/kotlin/app/revanced/patches/youtube/video/quality/RememberVideoQualityPatch.kt [hunk 4]
```diff
@@ -28,18 +25,25 @@ import com.android.tools.smali.dexlib2.iface.instruction.ReferenceInstruction
...
+)
+@Suppress("unused")
+object RememberVideoQualityPatch : BytecodePatch(
+    setOf(
         VideoQualitySetterFingerprint,
         VideoQualityItemOnClickParentFingerprint,
         NewVideoQualityChangedFingerprint
     )
 ) {
+    const val INTEGRATIONS_CLASS_DESCRIPTOR =
```
Comment (oSumAtrIX) [inline src/main/kotlin/app/revanced/patches/youtube/video/quality/RememberVideoQualityPatch.kt:44]: Can be private

File: src/main/kotlin/app/revanced/patches/youtube/video/videoqualitymenu/OldVideoQualityMenuResourcePatch.kt [hunk 5]
```diff
@@ -1,16 +1,20 @@
...
 import app.revanced.patches.shared.settings.preference.impl.StringResource
 import app.revanced.patches.shared.settings.preference.impl.SwitchPreference
-import app.revanced.patches.youtube.misc.settings.bytecode.patch.SettingsPatch
+import app.revanced.patches.youtube.misc.settings.SettingsPatch
+
+@Patch(
+    dependencies = [SettingsPatch::class, ResourceMappingPatch::class]
+)
+object OldVideoQualityMenuResourcePatch : ResourcePatch() {
+    var videoQualityBottomSheetListFragmentTitle = -1L
```
Comment (oSumAtrIX) [inline src/main/kotlin/app/revanced/patches/youtube/video/videoqualitymenu/OldVideoQualityMenuResourcePatch.kt:16]: Should be internal

File: src/main/kotlin/app/revanced/patches/youtube/layout/theme/ThemeBytecodePatch.kt [hunk 6]
```diff
@@ -0,0 +1,39 @@
...
+
+import app.revanced.patcher.data.BytecodeContext
+import app.revanced.patcher.patch.*
+import app.revanced.patcher.patch.annotation.CompatiblePackage
+import app.revanced.patcher.patch.annotation.Patch
+import app.revanced.patcher.patch.options.types.StringPatchOption.Companion.stringPatchOption
+import app.revanced.patches.youtube.layout.seekbar.bytecode.patch.SeekbarColorBytecodePatch
+
+@Patch(
+    name = "Theme",
```
[the remark this answers is not in the document]
Comment (Ushie) [reply src/main/kotlin/app/revanced/patches/youtube/layout/theme/ThemeBytecodePatch.kt:11]: Isn't this the actual patch?

[the remark this answers is not in the document]
Comment (oSumAtrIX) [reply src/main/kotlin/app/revanced/patches/youtube/layout/theme/ThemeBytecodePatch.kt:11]: You are right

**Pull request merged**